jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -257,21 +266,19 @@ object XGBoost extends Serializable {
    * @param useExternalMemory indicate whether to use external memory cache, by setting this flag as
    *                          true, the user may save the RAM cost for running XGBoost within Spark
    * @param missing the value represented the missing value in the dataset
-   * @param baseMargin initial prediction for boosting.
    * @throws ml.dmlc.xgboost4j.java.XGBoostError when the model training is failed
    * @return XGBoostModel when successful training
    */
   @throws(classOf[XGBoostError])
   def trainWithRDD(
-      trainingData: RDD[MLLabeledPoint],
+      trainingData: RDD[XGBLabeledPoint],
```

Comment:
So, to summarize you propose to:

1. Remove `baseMargin` from the deprecated `XGBoost.train`.
2. Make `XGBoost.trainWithRDD` private and keep `baseMargin` there.
3. Officially declare data frames as the only supported source of data for `XGBoost`.

On my side, I have mixed feelings about forcing the `DataFrame` API. Partly due to the fact that the app I'm working on cannot use `DataFrames` (yet), and partly because users having the input as RDD would do a double conversion behind the scenes RDD->DF->RDD. 

To me the real problem we are having is not RDD vs DataFrame but rather a lack of a deprecation process which, I think, could be tackled separately (see #1807 and #2553).